src/main/java/org/jenkinsci/plugins/DependencyTrack/DependencyTrackPublisher.java

```diff
@@ -178,91 +179,90 @@ public void perform(@Nonnull final Run<?, ?> build,
 
         if (StringUtils.isBlank(artifact)) {
             logger.log(Messages.Builder_Artifact_Unspecified());
-            build.setResult(Result.FAILURE);
-            return;
+            throw new AbortException("Artifact not specified");
```

Comment:
similar pattern here and below.  Rather than setting build status and not informing the pipeline flow, throw exceptions instead.